bite-size-notes/week-9-d-CKF.md

```diff
@@ -0,0 +1,136 @@
+
+Just like for discrete systems, we have the predict equations given as:
+$$\mu_{t}^{p}=(I+A\Delta t)\mu_{t-1} + B\Delta t u_{t-1}$$
+$$\Sigma_{t}^{p}=(I+A\Delta t)\Sigma_{t-1}(I+A\Delta t)^{T}+Q\Delta t$$
+
+Update equations are as follows:
+$$K\_{t}=\\Sigma\_{t}^p C^{T} (C \\Sigma\_{t}^{p} C^{T} + \\frac{1}{\\Delta t}R)^{-1}$$
+$$\mu_{t}=\mu_{t}^{p}+K_{t}(y_{t}-C\mu_{t}^{p})$$
+$$\Sigma_{t}=(I-K_{t}C)\Sigma_{t}^{p}$$
+
+Rearranging the gain equation we get,
+$$\\frac{1}{\\Delta t}K\_{t} = \\Sigma\_{t}^{p} C^{T} (C \\Sigma\_{t}^{p} C^{T} \\Delta t +R)^{-1}$$
+
+as $\Delta t$ goes to zero, the term on right goes to a finite value of $\Sigma_{t}^{p} C^{T} R^{-1}$. The only way, left hand side can be finite is when $K_t$ also goes to zero.
+This suggests that when continuous-time dynamics are approximated by
+discrete-time dynamics using a small sampling time, the Kalman gain
+approaches zero. Consequently, employing a small sampling time in
+discrete-time dynamics may not be viable.
+
+### Finite difference version of $Σ$
+
+Substituting the update equations into the predict equations for *Σ*, we
+get,
+$$\\Sigma\_{t}^{p}=(I+A\\Delta t)(I-K\_{t-1}C)\\Sigma\_{t-1}^{p}(I+A\\Delta t)^{T}+Q\\Delta t$$
+Rearranging this terms will give
+
+$$\\frac{1}{\\Delta t}(\\Sigma_t^p-\\Sigma\_{t-1}^p) = -\\frac{k\_{t-1}C\\Sigma\_{t-1}^p}{\\Delta t} + (A\\Sigma\_{t-1}^p+AK\_{k-1}C\\Sigma\_{t-1}^p+\\Sigma\_{t-1}^pA^T-K\_{t-1}C\\Sigma\_{t-1}^pA^T+Q) + O(\\Delta t^2)$$
+
+With limit, $Δt → 0$
+
+We get the finite difference version,
+$$\\underbrace{\\dot{\\Sigma}(t)=A\\Sigma(t)+\\Sigma(t)A^{T}-\\Sigma(t)C^{T}R^{-1}C\\Sigma(t)+Q}\_{\\text{Riccati ODE}}$$
+
+### Duality in the Riccati Equation
+
+CKF: 
+$$\dot{\Sigma}=A\Sigma+\Sigma A^{T}-\Sigma C^{T}R^{-1}C\Sigma+Q$$
+LQR:  
+$$-\dot{P}=A^{T} P+P A-P B R^{-1} B^{T} P+Q$$
+
+|                   **LQR**                   |              **CKF**              |
+|:-------------------------------------------:|:---------------------------------:|
+|                     $A$                     |             $A^{T}$               |
+|                     $B$                     |             $C^{T}$               |
+|             $Q_{\text{cost}}$               |           $Q=cov(w)$              |
+|             $R_{\text{cost}}$              |             $R=cov(v)$             |
+|               $t$ (backward)                | $t_{f}=t$\ (forward)              |
+| $P$ from $V(x)=x^{T}Px$ |      $Σ$  Covariance of Est.      |
+
+Differences between LQR and KF.
+
+### Finite Difference version of *μ*
+
+We can also carry out similar operations to arrive a finite difference equation for $\mu$,
+$$\mu_{t} = \mu_{t}^{p} + K_{t}(y_{t}-C\mu_{t}^{p})$$
+Plugging this into predict equation will give,
+$$\mu_{t}=(I+A\Delta t)\mu_{t-1}+K_{t}(y_{t}-C(I+A\Delta t)\mu_{t-1} + B\Delta tu_{t-1}))$$
```

Comment:
When I try to substitute the update equation for $\mu$ into the predict equation for $\mu^p$ I get:
$$\mu^p_t =  (I+\Delta t A)(\mu^p_{t-1} + K_{t-1}(y_{t-1} - C \mu^p_{t-1})) + \Delta t B u_{t-1}$$

The simplification from there to the final representation of $\dot{\mu}$ is difficult and involves substituting in the definition of K.  Adding these steps would be very helpful.

Reply:
I see. We do not need to do this calculation actually. We can write the equation just by combining update and predict equations.